### ISSUE TITLE
Optix Error: 'Invalid value'

### ISSUE THREAD

**Issue by matthew-wave** [zju3dv/LiDAR-RT#13]
Thank you for your work! When I am installing the third-party package diff-lidar-tracker, it shows that the installation is successful. There are also forwards.ptx and backwards.ptx in the/root/models/LiDAR RT main/submodules/diff lidar tracker/build/ptx directory. But it still keeps showing during training/ root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value' This error only produces blue images every 100 epochs.
My configuration is CUDA12.1, torch==2.3.1 torch vision==0.18.1 torch studio==2.3.1, and my GPU is A100-40GB.

Here are all the log outputs:
2025-06-18 06:11:46.739580: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-06-18 06:11:46.755486: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:467] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1750227106.772686  570319 cuda_dnn.cc:8579] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1750227106.777950  570319 cuda_blas.cc:1407] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
W0000 00:00:1750227106.792576  570319 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.
W0000 00:00:1750227106.792607  570319 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.
W0000 00:00:1750227106.792613  570319 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.
W0000 00:00:1750227106.792618  570319 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.
2025-06-18 06:11:46.797113: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
W0000 00:00:1750227109.032728  570319 gpu_device.cc:2341] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
Optix Log[2][COMPILER]: 'COMPILE ERROR: Module compilation failed
Info: Module Statistics
        payload values        :          2
        attribute values      :          0
        Pipeline configuration:  (default)
Info: Properties for entry function "__anyhit__ot"
        semantic type                :                 ANYHIT
        trace call(s)                :                      0
        continuation callable call(s):                      0
        direct callable call(s)      :                      0
        basic block(s)               :                     34
        instruction(s)               :                    302
Info: Compiled Module Summary
        non-entry function(s):     0
        basic block(s)       :     0
        instruction(s)       :     0
'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:65 Optix Error: 'Optix Log[2][COMPILER]: 'COMPILE ERROR: Module compilation failed
Info: Module Statistics
        payload values        :          2
        attribute values      :          0
        Pipeline configuration:  (default)
Info: Properties for entry function "__anyhit__ot"
        semantic type                :                 ANYHIT
        trace call(s)                :                      0
        continuation callable call(s):                      0
        direct callable call(s)      :                      0
        basic block(s)               :                     34
        instruction(s)               :                    302
Info: Compiled Module Summary
        non-entry function(s):     0
        basic block(s)       :     0
        instruction(s)       :     0
'
Internal compiler error'
Optix Log[2][COMPILER]: 'COMPILE ERROR: programDescriptions[0].raygen.module argument is not an optixProgramGroupCreate
'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:80 Optix Error: 'Optix Log[2][COMPILER]: 'COMPILE ERROR: programDescriptions[0].raygen.module argument is not an optixProgramGroupCreate
'
Invalid value'
Optix Log[2][COMPILER]: 'COMPILE ERROR: programDescriptions[0].hitgroup.moduleAH argument is not an optixProgramGroupCreate
'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:98 Optix Error: 'Optix Log[2][COMPILER]: 'COMPILE ERROR: programDescriptions[0].hitgroup.moduleAH argument is not an optixProgramGroupCreate
'
Invalid value'
Optix Log[2][COMPILER]: 'COMPILE ERROR: programGroups[0] argument is null
'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:111 Optix Error: 'Optix Log[2][COMPILER]: 'COMPILE ERROR: programGroups[0] argument is null
'
Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:118 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:118 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:129 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:141 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:161 Optix Error: 'Invalid value'
Optix Log[2][COMPILER]: 'COMPILE ERROR: Module compilation failed
Info: Module Statistics
        payload values        :          2
        attribute values      :          0
        Pipeline configuration:  (default)
Info: Properties for entry function "__anyhit__ot"
        semantic type                :                 ANYHIT
        trace call(s)                :                      0
        continuation callable call(s):                      0
        direct callable call(s)      :                      0
        basic block(s)               :                     34
        instruction(s)               :                    302
Info: Compiled Module Summary
        non-entry function(s):     0
        basic block(s)       :     0
        instruction(s)       :     0
'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:65 Optix Error: 'Optix Log[2][COMPILER]: 'COMPILE ERROR: Module compilation failed
Info: Module Statistics
        payload values        :          2
        attribute values      :          0
        Pipeline configuration:  (default)
Info: Properties for entry function "__anyhit__ot"
        semantic type                :                 ANYHIT
        trace call(s)                :                      0
        continuation callable call(s):                      0
        direct callable call(s)      :                      0
        basic block(s)               :                     34
        instruction(s)               :                    302
Info: Compiled Module Summary
        non-entry function(s):     0
        basic block(s)       :     0
        instruction(s)       :     0
'
Internal compiler error'
Optix Log[2][COMPILER]: 'COMPILE ERROR: programDescriptions[0].raygen.module argument is not an optixProgramGroupCreate
'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:80 Optix Error: 'Optix Log[2][COMPILER]: 'COMPILE ERROR: programDescriptions[0].raygen.module argument is not an optixProgramGroupCreate
'
Invalid value'
Optix Log[2][COMPILER]: 'COMPILE ERROR: programDescriptions[0].hitgroup.moduleAH argument is not an optixProgramGroupCreate
'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:98 Optix Error: 'Optix Log[2][COMPILER]: 'COMPILE ERROR: programDescriptions[0].hitgroup.moduleAH argument is not an optixProgramGroupCreate
'
Invalid value'
Optix Log[2][COMPILER]: 'COMPILE ERROR: programGroups[0] argument is null
'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:111 Optix Error: 'Optix Log[2][COMPILER]: 'COMPILE ERROR: programGroups[0] argument is null
'
Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:118 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:118 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:129 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:141 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/optix_tracer/optix_wrapper.cpp:161 Optix Error: 'Invalid value'
Jitting Chamfer 3D
Loaded JIT 3D CUDA chamfer distance

====== [Loading] Waymo Open Dataset ======
2025-06-18 06:12:13.467232: I tensorflow/core/kernels/data/tf_record_dataset_op.cc:381] TFRecordDataset `buffer_size` is unspecified, default to 262144
2025-06-18 06:14:33.348386: I tensorflow/core/framework/local_rendezvous.cc:407] Local rendezvous is aborting with status: OUT_OF_RANGE: End of sequence
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:22<00:00,  2.21it/s]
------------
train: [148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 159, 160, 161, 162, 163, 164, 165, 166, 167, 169, 170, 171, 172, 173, 174, 175, 176, 177, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196]
eval: [158, 168, 178, 197]
[Loaded] background guassians
Number of points at initialization:  240075
Number of points at initialization:  10000
Number of points at initialization:  10000
Number of points at initialization:  10000
[Loaded] object guassians
Task: default, Experiment: test, Scene: wd1
Output dir:  /data2/lidar-rt/output/default/test/scene_wd1
Continuing from iteration  0
Training progress:   0%|                                                                                                                                                                                                                                                                | 0/30000 [00:00<?, ?it/s]/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:256 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:256 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:256 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:256 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:256 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:256 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:256 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:256 Optix Error: 'Invalid value'
/root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/trace_surfels.cpp:378 Optix Error: 'Invalid value'

**Comment by matthew-wave** [comment]
My system is ubnutu20.04

**Comment by Honuaye** [comment]
Ensure that your Nvidia driver is unique and confl

**Comment by matthew-wave** [comment]
> Ensure that your Nvidia driver is unique and conflict free

Thank you for your reply. My current driver is 550.54.14 and I have confirmed that there are no conflicts. The output of nvcc -V is CUDA12.1. As far as I know, this meets the environment requirements of lidar-rt. How can we solve this problem. I also followed other reply methods and ran export TORCH-CUDA_ARCH_LIST="8.0" before following diff-lidar-tracker, but it still doesn't work.

**Comment by cxzhou35** [comment]
This annoying error sometimes pop up randomly and perhaps go away after the restart. 
Unfortunately we do not have the A100 GPU to debug such issue, here are some advices that might be helpful for you.

1. If all of the tested scenes were failed to work, you might need to reinstall the `diff-lidar-tracker` module and check whether the [minimal example](https://github.com/zju3dv/LiDAR-RT/issues/5#issuecomment-2826225443) works.
Add the `-v` flag during reinstall to get more verbose messages: `pip install -v ./submodules/diff-lidar-tracer`.

2. Recheck the environments. Below are our testing environments, just for reference:
```sh
os: ubuntu 24.04
nvcc: cuda 12.1
nvidia driver: 535.183.01
python: 3.11.9
torch: 2.3.1
```

3. Adjust the OptiX version, I haven't tried it.

**Comment by matthew-wave** [comment]
> This annoying error sometimes pop up randomly and perhaps go away after the restart. Unfortunately we do not have the A100 GPU to debug such issue, here are some advices that might be helpful for you.
> 
> 1. If all of the tested scenes were failed to work, you might need to reinstall the `diff-lidar-tracker` module and check whether the [minimal example](https://github.com/zju3dv/LiDAR-RT/issues/5#issuecomment-2826225443) works.
>    Add the `-v` flag during reinstall to get more verbose messages: `pip install -v ./submodules/diff-lidar-tracer`.
> 2. Recheck the environments. Below are our testing environments, just for reference:
> 
> os: ubuntu 24.04
> nvcc: cuda 12.1
> nvidia driver: 535.183.01
> python: 3.11.9
> torch: 2.3.1
> 3. Adjust the OptiX version, I haven't tried it.

I see/ root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/third_party/optix/include Do I still need to install Optix7.0 locally first, as there is the source code for Optix/Include pulled from GitHub in the directory?
I tried not to use the three versions of Optix 7.7, 7.2, and 8.0, but the method I tried was to/ root/models/LiDAR-RT-main/submodules/diff-lidar-tracer/third_party/optix/include Replace the source code with a new version, I'm not sure if this is correct. I will still encounter the same error when testing with the above methods.

**Comment by matthew-wave** [comment]
congratulate! I have solved this problem, and the final configuration I used is 3070ti, CUDA11.8, torch 2.3.1+cu118, optix8, Ubuntu 20.04. Before this, I tried various version combinations on 5090D and A100 but were unsuccessful.
I am currently training 30000 times according to the configuration, but the program was killed at 19000 times (I think this reason is related to the fact that 3070ti only has 8G). I am inferring the generated point cloud using the model trained 19000 times, and then drawing the point cloud using the provided viewer.py. This image can only be enlarged to this scale. Is this normal? Is this situation due to insufficient training sessions.

![Image](https://github.com/user-attachments/assets/ac2ffd78-d4ac-4e4c-88a5-37ea5e2cf49c)

**Comment by cxzhou35** [comment]
Sorry for the late reply. I am glad to know you have solved the annoying setup issue.

Yeah, there will be some outlier points in the scene, you can filter these points by the distance to the scene centers as the mask.
But it's weird to see the problem that about 20% points at the origin. What's the scene you test on?

**Comment by matthew-wave** [comment]
The scenario I trained on is Waymo/dynamic/3, but the results of training with Waymo/dynamic/1 also show a large number of outliers. The default configuration used for training was 3000 times, and the optimal output model was obtained at the 2000th training session.
The script used to draw point cloud maps is viewer.py.
